blog/2021-03-16-BatchSolvingWithAmq.adoc

```diff
@@ -0,0 +1,182 @@
+:awestruct-share_image_filename: architecture.png
+
+Let's take a look at a new https://github.com/kiegroup/optaplanner-quickstarts/tree/stable/amq-quarkus-school-timetabling[quickstart], which shows how to build a scalable, stateless service capable of solving numerous instances of the same planning problem.
+This post will be most interesting to developers looking for how to bring OptaPlanner from an embedded engine to a service communicating with other applications.
+
+== Batch Solving
+
+Very often, there are multiple instances of the same planning problem to solve.
+Either these come from splitting enormous input problem into smaller pieces, or just from the need to solve completely unrelated data sets.
+Imagine scheduling routes of vehicles independently per regions, or optimizing school timetables for numerous schools.
+To take advantage of time, you run OptaPlanner every night to prepare for the next day in the business or even longer for the next semester.
+On the other hand, during the day or in the middle of the semester, there is nothing to optimize and so there should be no OptaPlanner running.
+In other words, these cases call for a batch solving.
+
+== School Timetabling
+
+The quickstart focuses on the school timetabling problem, which is described in depth in the https://quarkus.io/guides/optaplanner[Quarkus guide].
+Let’s just very briefly revisit the problem domain and its constraints.
+
+In the school timetabling problem, the goal is to assign each lesson to a room and a timeslot.
+To use the OptaPlanner vocabulary, the `Lesson` is a planning entity and its references to the `Room` and the `Timeslot` are planning variables.
+
+image::schoolTimetablingClassDiagram.png[School Timetabling class diagram]
+
+The `TimeTableConstraintProvider` defines the following constraints on how the lessons should be assigned to timeslots and rooms:
+
+* A room can have at most one lesson at the same time (hard).
+* A teacher can teach at most one lesson at the same time (hard).
+* A student can attend at most one lesson at the same time (hard).
+* A teacher prefers to teach in a single room (soft).
+* A teacher prefers to teach sequential lessons and dislikes gaps between lessons (soft).
+* A student dislikes sequential lessons on the same subject (soft).
+
+== Quickstart structure
+
+image::projectStructure.png[Project structure]
+
+The project consists of three modules:
+
+* *amq-quarkus-school-timetabling-common* defines the problem domain, and the `SolverRequest` and `SolverResponse` classes for messaging.
+The following two modules depend on this one.
+* *amq-quarkus-school-timetabling-client* is the Client Quarkus application that contains a UI, a REST endpoint and a demo data generator.
+* *amq-quarkus-school-timetabling-solver* is the Solver Server Quarkus application that solves school timetabling problem instances coming via a message queue `solver_request`.
+
+== Messaging
+
+image::architecture.png[Architecture]
```

Comment:
Can the architecture diagram show multiple solver nodes? (and maybe multiple client nodes).
"multiple" can just be 2 extra rounded rectangle under the existing rounded rectangle for space efficiency and clarity.

Reply:
I welcome every opportunity to improve my Inkscape skills :)